src/rpdk/core/extensions.py

```diff
@@ -0,0 +1,10 @@
+from .plugin_registry import get_extensions
+
+
+def setup_subparsers(subparsers, parents):
+    extensions = get_extensions()
+
+    for extension_cls in extensions.values():
+        extension = extension_cls()()
+        parser = subparsers.add_parser(extension.command_name, parents=parents)
+        extension.setup_parser(parser)
```

Comment:
What can the extension do given the subparser?

Can it access the parent from the subparse and start mutating the parent in unexpected ways?

Reply:
From `argparse` [documentation](https://docs.python.org/3/library/argparse.html#argparse.ArgumentParser.add_subparsers), it doesn't look like a subparser can access the parent parser. The closest is the `parents` argument which just tells the subparser to inherit the parents' arguments as well (this is something we might want to remove for this case, actually).